pkg/user/external_server.go

```diff
@@ -0,0 +1,263 @@
+package user
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func readRequestBody(req *http.Request, body interface{}) error {
+	if err := json.NewDecoder(req.Body).Decode(body); err != nil {
+		return err
+	}
+	return nil
+}
+
+func sendResponse(rw http.ResponseWriter, status int, body interface{}) {
+	rw.Header().Set("Content-Type", "application/json")
+	rw.WriteHeader(status)
+
+	if body != nil {
+		// Set up the pipe to write data directly into the Reader.
+		pr, pw := io.Pipe()
+
+		// Write JSON-encoded data to the Writer end of the pipe.
+		// Write in a separate concurrent goroutine, and remember
+		// to Close the PipeWriter, to signal to the paired PipeReader
+		// that we’re done writing.
+		go func() {
+			pw.CloseWithError(json.NewEncoder(pw).Encode(body))
+		}()
+
+		_, err := io.Copy(rw, pr)
+		if err != nil {
+			log.WithError(err).Warn("[ExternalServer] error writing response")
+		}
+	}
+}
+
+func StartExternalServer(listenAddr string, stop <-chan struct{}, um UserManager) error {
+	ctx := context.Background()
+	s := &http.Server{
+		Addr: listenAddr,
+		Handler: http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			l := log.WithField("path", req.URL.Path).WithField("method", req.Method)
+			l.Debug("[ExternalUserManagerServer] request received")
+			switch fmt.Sprintf("%s %s", req.Method, req.URL.Path) {
+			case "PUT /user/admin":
+				var u User
+				err := readRequestBody(req, &u)
+				if err != nil {
+					l.WithError(err).Error("[ExternalUserManagerServer] Bad request")
+					http.Error(rw, err.Error(), http.StatusBadRequest)
```

Comment:
here you can `return` so no need for next "else" but generally it should live in a separate handler function